2024/day2/day2.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type Reports [][]int
+
+//go:embed input.txt
+var input string
+
+func main() {
+	reports := parseInput(input)
+	fmt.Printf("reports: %v\n", reports)
+
+	safeReports := SafeReports(reports)
+	fmt.Printf("Safe Reports Len: %d\n", safeReports)
+
+}
+
+func SafeReports(reports Reports) int {
+	r := 0
+	for _, report := range reports {
+		if IsReportSafe(report) || IsLessSafer(report) {
+			fmt.Printf("This is safe : %v\n", report)
+			r++
+		}
+	}
+
+	return r
+}
+
+func IsReportSafe(report []int) bool {
+	return checkIncrease(report) || checkDecrease(report)
```

Comment:
The `for` loops in checkIncrease and checkDecrease are the same and can be moved to this function to only call it once instead of twice.

Reply:
Agreed. Kept it for clarity.